downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md

```diff
@@ -0,0 +1,468 @@
+- Changed to return all the errors related to a GraphQL document in a single response
+
+#### `gRPC` package
+
+- Updated Protocol Buffers version to 3.21.7
+
+### Bug fixes
+
+To view bug fixes, see the [GitHub milestone for 2201.3.0 (Swan Lake)](https://github.com/ballerina-platform/ballerina-standard-library/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+is%3Aclosed).
+
+## Developer tools updates
+
+### New features
+
+#### Language Server
+
+- Introduced a new `Extract to local variable` code action
+- Introduced a new `Extract to constant` code action
+- Introduced a new `Extract to function` code action
+- Introduced a new code action to convert an array to array mapping into a query expression
+- Introduced the `loadProject()` API to the workspace manager to open up a project programmatically
+
+#### CLI
+
+##### Native-image build (Experimental)
+
+Introduced the `--native` flag, which generates a GraalVM native executable when building a Ballerina project.
+
+- The generated executable contains the modules in the current package, their dependencies, Ballerina runtime, and statically linked native code from the JDK
+
+- The new CLI command will be `bal build --native`
+
+>**Note:** There is a known [issue](https://github.com/ballerina-platform/ballerina-lang/issues/38665) in the native image build when using locks along with loops. 
+
+#### Debugger
+
+Introduced the `run-in-terminal` feature, which provides the capability to debug Ballerina programs that take user inputs in the launch mode. This can be achieved by setting the `terminal` attribute in the launch configurations to `integrated`. Once the configuration is set, launching the debugger will cause the Ballerina programs to be executed in a separate integrated VS Code terminal.
+
+### Improvements
+
+#### Language Server
+
+- Improved completion sorting within the expression of the `if` condition
+- Improved the `Change variable type` code action
+
+#### OpenAPI Tool
+
+Improved CLI client generation to generate the Ballerina client resource methods by default. (However, the users can use `--client-methods=remote` command option to switch back to client remote functions instead.)
+
+#### GraphQL Tool
+
+- Introduced single client generation for a single schema definition
+- Improved the GraphQL client config generation to support runtime configurability
+
+### Bug fixes
+
+To view bug fixes, see the GitHub milestone for 2201.3.0 (Swan Lake) of the repositories below.
+
+- [Language Server](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+label%3ATeam%2FLanguageServer)
+- [OpenAPI](https://github.com/ballerina-platform/openapi-tools/issues?q=is%3Aissue+milestone%3A%22Swan+Lake+2201.3.0%22+label%3AType%2FBug+is%3Aclosed)
+- [Semantic API](https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AArea%2FSemanticAPI+)
```

Comment:
Semantic API improvements and new features are listed under the main title of `Compiler API updates`. (https://github.com/ballerina-platform/ballerina-dev-website/blob/3ccd52ea52812ecf78b1b2072078893243971d7a/downloads/swan-lake-release-notes/swan-lake-2201.3.0/RELEASE_NOTE.md#compiler-api-updates) Then why are we adding only the bug fixes under `Developer tools updates` section? Also link is incorrect as it is not filtered by bugs. 

@praneesha / @mohanvive

Reply:
@mohanvive - What is the difference between the 2 links below:

1. https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AArea%2FSemanticAPI+
2. https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FCompilerFETools+

Which one should we put as the bug fixes milestone link under the `Compiler API updates` section?

Reply:
`CompilerFETools` label is a broader one. `SemancticAPI` is just one of the compiler API tools.
It seems, under the compiler API updates we list all kinds of compiler API changes. Not only the Semantic API. Hence the filter should consider `CompilerFETools`. Further. since we are planning to list down the bugs we could use that specific label as well. Hence, you could use the below one.

https://github.com/ballerina-platform/ballerina-lang/issues?q=is%3Aissue+milestone%3A2201.3.0+label%3AType%2FBug+is%3Aclosed+label%3ATeam%2FCompilerFETools

Reply:
Updated.